<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>998</width>
    <height>655</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>语涵编译器</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="2" column="0">
     <widget class="QGroupBox" name="miscSettingsGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>其他设置</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="2" column="0">
        <widget class="QCheckBox" name="languageCheckBox">
         <property name="text">
          <string>指定语言</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QCheckBox" name="verboseCheckBox">
         <property name="text">
          <string>显示详细信息</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="4">
        <widget class="FileSelectionWidget" name="mainExecutableSelectionWidget" native="true"/>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="languageComboBox"/>
       </item>
       <item row="2" column="3">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1" colspan="4">
        <widget class="FileSelectionWidget" name="pluginSelectionWidget" native="true"/>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>主程序路径</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>插件目录</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="4">
        <widget class="QCheckBox" name="autoSearchPathCheckBox">
         <property name="text">
          <string>添加输入文档时自动添加所在目录为搜索目录</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="outputGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>输出选项</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QTabWidget" name="outputSelectionTabWidget">
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="noOutputTab">
          <attribute name="title">
           <string>无输出</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>（无输出时此处无选项）</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="renpyOutputTab">
          <attribute name="title">
           <string>RenPy</string>
          </attribute>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>输出到 game 目录</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="renpyUseTemplateCheckBox">
             <property name="text">
              <string>使用模板</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="FileSelectionWidget" name="renpyTemplateSelectionWidget" native="true"/>
           </item>
           <item row="0" column="1">
            <widget class="FileSelectionWidget" name="renpyOutputDirWidget" native="true"/>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="webgalOutputTab">
          <attribute name="title">
           <string>WebGal</string>
          </attribute>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>输出到 game 目录</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="FileSelectionWidget" name="webgalOutputDirWidget" native="true"/>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="webgalUseTemplateCheckBox">
             <property name="text">
              <string>使用模板</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="FileSelectionWidget" name="webgalTemplateSelectionWidget" native="true"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="inputGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>输入选项</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0" rowspan="2">
        <widget class="FileListInputWidget" name="inputWidget" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="1" rowspan="2">
        <widget class="FileListInputWidget" name="searchPathInputWidget" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="3">
     <widget class="QGroupBox" name="executeGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>执行</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QPlainTextEdit" name="commandTextEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>100</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="executePushButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>执行</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" rowspan="3">
     <widget class="QGroupBox" name="passGroupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>优化、分析选项</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QCheckBox" name="entryInferenceCheckBox">
           <property name="toolTip">
            <string>对于 RenPy 输出来说，这个选项会加 label start, 内容为跳转至第一个没有被调用的函数、章节入口处</string>
           </property>
           <property name="text">
            <string>添加起始跳转</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="longSaySplittinggroupBox">
         <property name="toolTip">
          <string>如果某句发言内容过长，启用该优化后所有长于给定长度的发言都将被按标点符号进行分隔</string>
         </property>
         <property name="title">
          <string>长句分句</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="2">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>目标半宽字符数</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>半宽字符数（英文=1，汉字=2）达到该值时开始分隔：</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="minlenSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QSpinBox" name="targetlenSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="icfgGroupBox">
         <property name="title">
          <string>生成控制流图</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>保存到：</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="FileSelectionWidget" name="icfgExportSelectionWidget" native="true"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="assetUsageGroupBox">
         <property name="title">
          <string>资源使用情况分析</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_11">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>保存到：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="FileSelectionWidget" name="assetUsageSelectionWidget" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="sayDumpGroupBox">
         <property name="title">
          <string>导出发言信息</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>格式预设</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="sayDumpPresetComboBox"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>保存到：</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="FileSelectionWidget" name="sayDumpLocationSelectionWidget" native="true"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="customPassGroupBox">
         <property name="toolTip">
          <string>如果您在插件中自定义了分析或转换，可以在这加入对其的调用</string>
         </property>
         <property name="title">
          <string>自定义分析、转换</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_5">
          <item row="0" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>IR 阶段 (VNModel)</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="irPassLineEdit"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="translationImportGroupBox">
         <property name="title">
          <string>导入自定义翻译、别名</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_12">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>输入JSON文件：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="FileSelectionWidget" name="customTranslationImportLocationSelectionWidget" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>998</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="operationMenu">
    <property name="title">
     <string>操作</string>
    </property>
    <addaction name="actionTranslationExport"/>
   </widget>
   <widget class="QMenu" name="toolMenu">
    <property name="title">
     <string>工具</string>
    </property>
    <addaction name="actionImagePackTool"/>
    <addaction name="actionNewUI"/>
   </widget>
   <addaction name="operationMenu"/>
   <addaction name="toolMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionTranslationExport">
   <property name="text">
    <string>导出翻译、别名文件</string>
   </property>
  </action>
  <action name="actionImagePackTool">
   <property name="text">
    <string>图片包</string>
   </property>
  </action>
  <action name="actionNewUI">
   <property name="text">
    <string>新版UI</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FileSelectionWidget</class>
   <extends>QWidget</extends>
   <header>fileselectionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FileListInputWidget</class>
   <extends>QWidget</extends>
   <header>filelistinputwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
